src/cluster/placement/placement.go

```diff
@@ -311,6 +319,7 @@ func Validate(p Placement) error {
 	totalLeaving := 0
 	totalInit := 0
 	totalInitWithSourceID := 0
+	instancesLeavingShardsWithMatchingInitShards := make(map[string]map[uint32]string)
```

Comment:
nit; never uses the string value, maybe this should be a `make(map[string]map[uint32]struct{})`?

Reply:
We do to print out the error of which host previously had claimed that leaving shard to match it's own initializing shard:
```go
					match := matches[s.ID()]
					if match != "" {
						return fmt.Errorf(
							"instance %s has initializing shard %d with "+
								"source ID %s but leaving instance has shard already matched by %s",
							instance.ID(), s.ID(), sourceID, match)
					}
```